python/whylogs/experimental/core/metrics/udf_metric.py

```diff
@@ -20,37 +12,38 @@
 )
 from whylogs.core.metrics.multimetric import MultiMetric, SubmetricSchema
 from whylogs.core.preprocessing import PreprocessedColumn
+from whylogs.core.resolvers import STANDARD_RESOLVER, ResolverSpec, _allowed_metric
 
 logger = logging.getLogger(__name__)
 
 
-class DefaultSchema(SubmetricSchema):
+class DeclarativeSubmetricSchema(SubmetricSchema):
+    def __init__(self, resolvers: List[ResolverSpec], default_config: Optional[MetricConfig] = None) -> None:
+        self._default_config = default_config or MetricConfig()
+        self._resolvers = resolvers.copy()
+
     def resolve(self, name: str, why_type: DataType, fi_disabled: bool = False) -> Dict[str, Metric]:
-        metrics: Dict[str, Metric] = {
-            "counts": StandardMetric.counts.zero(MetricConfig()),
-            "types": StandardMetric.types.zero(MetricConfig()),
-            "cardinality": StandardMetric.cardinality.zero(MetricConfig()),
-        }
+        result: Dict[str, Metric] = {}
+        for resolver_spec in self._resolvers:
+            col_name, col_type = resolver_spec.column_name, resolver_spec.column_type
+            if (col_name and col_name == name) or (col_name is None and isinstance(why_type, col_type)):  # type: ignore
+                for spec in resolver_spec.metrics:
+                    config = spec.config or self._default_config
+                    if _allowed_metric(config, spec.metric):
+                        result[spec.metric.get_namespace()] = spec.metric.zero(config)
+
+        return result
 
-        if isinstance(why_type, Integral):
-            metrics["distribution"] = StandardMetric.distribution.zero(MetricConfig())
-            metrics["ints"] = StandardMetric.ints.zero(MetricConfig())
-            if not fi_disabled:
-                metrics["frequent_items"] = StandardMetric.frequent_items.zero(MetricConfig())
-        elif isinstance(why_type, Fractional):
-            metrics["distribution"] = StandardMetric.distribution.zero(MetricConfig())
-        elif isinstance(why_type, String) and not fi_disabled:
-            metrics["frequent_items"] = StandardMetric.frequent_items.zero(MetricConfig())
 
-        return metrics
+DefaultSchema = DeclarativeSubmetricSchema(STANDARD_RESOLVER)
```

Comment:
Its surprising and inconsistent styling for `DefaultSchema` to now be an instance of a schema rather than a class.

You could replace DefaultSchema with a factory for creating instances of default schemas?

Or to preserve existing functionality use `functools.partial`:
```suggestion
DefaultSchema = functools.partial(DeclarativeSubmetricSchema, resolvers=STANDARD_RESOLVER)
```